src/main/java/org/folio/rest/impl/CheckinReceivePiecesHelper.java

```diff
@@ -737,4 +746,27 @@ private boolean isMissingLocation(PoLine poLine, Piece piece) {
       }
     });
   }
+
+  CompletableFuture<Void> removeForbiddenEntities(List<PoLine> poLines, Map<String, Map<String, T>> pieces) {
+    if(!poLines.isEmpty()) {
+      Map<String, List<PoLine>> poLinesGroupedByOrderId = poLines.stream().collect(groupingBy(PoLine::getPurchaseOrderId));
+      String query = buildQuery(convertIdsToCqlQuery(poLinesGroupedByOrderId.keySet()), logger);
+      String url = String.format(GET_PURCHASE_ORDERS, poLinesGroupedByOrderId.size(), 0, query, lang);
+      return CompletableFuture.allOf(handleGetRequest(url, httpClient, ctx, okapiHeaders, logger)
+        .thenApply(json -> json.mapTo(PurchaseOrders.class).getPurchaseOrders())
+        .thenApply(orders -> orders.stream().map(order -> protectionHelper.isOperationRestricted(order.getAcqUnitIds(), ProtectedOperationType.UPDATE)
```

Comment:
`CompletableFuture.allOf` should be used for list of futures prepared by the stream. Currently you have structure `CompletableFuture.allOf(CompletableFuture<List<CompletableFuture<Void>>)` which actually means `allOf` will return completed future right away and the logic will not wait for list of futures completion at all...
You can move this logic to separate method to improve readability...